ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt

```diff
@@ -48,6 +49,8 @@ public class CIOApplicationEngine(
     private val startupJob: CompletableDeferred<Unit> = CompletableDeferred()
     private val stopRequest: CompletableJob = Job()
 
+    override val networkAddresses: Deferred<List<NetworkAddress>> = CompletableDeferred()
```

Comment:
I would add the `suspend` method instead: deferred expose `cancel` and other public calls

Reply:
good point, updated